manifests/bucketeer/charts/api/templates/deployment.yaml

```diff
@@ -172,9 +181,15 @@ spec:
             preStop:
               exec:
                 command:
-                  - "/bin/sh"
-                  - "-c"
-                  - "wget -O- --post-data='{}' http://localhost:$ENVOY_ADMIN_PORT/healthcheck/fail; while [ $(netstat -plunt | grep tcp | grep -v envoy | wc -l) -ne 0 ]; do sleep 1; done;"
+                  - /bin/sh
+                  - -c
+                  - |
+                    admin_port={{ .Values.envoy.adminPort }}
+                    # Wait for load balancer propagation (must match app container propagation delay)
+                    sleep 15
+                    wget -q -T 1 -O- --method=POST --body-data='' \
+                      "http://localhost:${admin_port}/drain_listeners?graceful" || true
+                    exit 0
```

Comment:
This will ensure that the envoy container waits until the LB stops sending new requests before shutting down gracefully the in-flight requests.